### ISSUE TITLE
Provider ghe_base_url variable doesn't work in 4.14.0 version

### ISSUE THREAD

**Issue by rafajurbano** [integrations/terraform-provider-github#903]
Hi,

When trying to define a different ghe_base_url, it keeps making requests to the github.com API.
In the following image to be able to see the error:
![image](https://user-images.githubusercontent.com/47658908/132849778-b0668d17-1e0c-46a6-8587-2693da827f60.png)

![image](https://user-images.githubusercontent.com/47658908/132849524-1759705c-5f88-4238-8236-5874de2be8b5.png)

Currently I am still working with version 4.13.0 because it does not produce this error. My intention is to inform so that it can be solved in future versions.

Thanks!

**Comment by jcudit** [comment]
Can you try with `GITHUB_BASE_URL` set in your shell instead of via the provider configuration block? Interested in seeing if this alternate method works.

We have plans to revamp provider authentication configuration so any additional information will help guide that.

**Comment by aidan-mundy** [comment]
I believe this error can happen when the `owner` is a blank string and the provided `base_url` is wrong. Check to make sure both of those values are being retrieved correctly.

**Comment by vanderboon** [comment]
I see this bug also.

Removing the `base_url` from the config and setting `GITHUB_BASE_URL` resolves it. 

With `base_url` set...
```
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: 2021/12/10 09:16:14 [DEBUG] Github API Request Details:
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: ---[ REQUEST ]---------------------------------------
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: GET /user HTTP/1.1
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Host: api.github.com
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: User-Agent: go-github
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Accept: application/vnd.github.v3+json
2021-12-10T09:16:14.023Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Accept-Encoding: gzip
```

With `GITHUB_BASE_URL` set...
```
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: 2021/12/10 09:14:57 [DEBUG] Github API Request Details:
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: ---[ REQUEST ]---------------------------------------
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: PATCH /api/v3/repos/MYOWNER/MYREPO HTTP/1.1
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Host: github.MYDOMAIN.com
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: User-Agent: go-github
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Content-Length: 415
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Accept: application/vnd.github.baptiste-preview+json, application/vnd.github.nebula-preview+json
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Content-Type: application/json
2021-12-10T09:14:57.733Z [DEBUG] plugin.terraform-provider-github_v4.18.2: Accept-Encoding: gzip
```

**Comment by tgpfeiffer** [comment]
For me even the oldest available release (4.2.0) of this provider shows the same behavior.
